Pylexibank2

tresoldi opened the pull request:
Moves to pylexibank2. Segmentation is not ready yet, but preliminary profiles are in /etc and will be added later.

Note that download is failing, as I kept the previous code. What it is the recommended way to download and unpack with pylexibank2?

Commit:
First pass at pylexibank2

Commit:
Explicit entrie in lexemes

Commit:
Removing new lines

Commit:
More newline replacements

Commit:
Removed debug code

Commit:
First pass at segmented data

Commit:
Updated list of contributors

Commit:
Removing code for profiles (to be added later)

Commit:
Removing stray newlines

SimonGreenhill commented on README.md:
```diff
@@ -21,5 +21,12 @@ Conceptlists in Concepticon:
+## Possible Improvements:
+
+- Languages linked to [bookkeeping languoids in Glottolog](http://glottolog.org/glottolog/glottologinformation#bookkeepinglanguoids):
+  - Zinacantán Tzotzil [tzot1264](http://glottolog.org/resource/languoid/id/tzot1264)
```
can we fix this while we're at it?

SimonGreenhill commented on cldf/languages.csv:
```diff
@@ -1,42 +1,42 @@
 ID,Name,Glottocode,Glottolog_Name,ISO639P3code,Macroarea,Latitude,Longitude,Family
-1,Swahili,swah1253,,swh,,,,
-2,Iraqw,iraq1241,,irk,,,,
```
can we try keep the ISO codes too please?

tresoldi replied:
Oh sorry, I removed them and the Tzotzil correction when playing with multiple profiles. I'll add them now.

LinguList commented on lexibank_wold.py:
```diff
@@ -24,41 +32,83 @@ class WOLDLexeme(Lexeme):
+        languages = self.raw_dir.read_csv("languages.csv", dicts=True)
+        for language in languages:
+            args.writer.add_language(
+                ID=language["ID"],
```
Do you NEED to set the ID as an ID here? I'd prefer to have an ID that reminds of the language name. It would be easier to check data afterwards.

LinguList commented on lexibank_wold.py:
```diff
@@ -24,41 +32,83 @@ class WOLDLexeme(Lexeme):
+        # add the languages from the language file
+        # NOTE: the source lists all languages, including proto-languages,
+        # but the `forms` only include the first 41 in the list
+        languages = self.raw_dir.read_csv("languages.csv", dicts=True)
```
Ah, sure, the languages should of course be placed into `etc/languages.csv` or is there any reason for not doing so?

tresoldi replied:
The raw source is a CLDF dataset, so that a second language file seemed redundant and there was an old discussion about keeping the same IDs (so not to have a mismatch between WOLD in itself and WOLD in Lexibank).

Should I override it?

LinguList replied:
I'd opt for overriding. I know we had the discussion, but I don't agree with keeping IDS, they could rather be kept in an extra column, as I did with STEDT IDs.

tresoldi replied:
Just to make sure: yes to override the IDs in code, but no to an `etc/languages.csv` file?

tresoldi replied:
Or an `etc/` file nonetheless so we can easily use `args.writer.add_languages()`?

LinguList replied:
As I sad: there should be an `etc/languages.csv` file.

chrzyki commented on lexibank_wold.py:
```diff
@@ -24,41 +32,83 @@ class WOLDLexeme(Lexeme):
-                    gc, iso = "tzot1259", "tzo"
-                if row["ID"] in vocab_ids:
-                    ds.add_language(ID=row["ID"], Name=row["Name"], Glottocode=gc, ISO639P3code=iso)
+        # add concepts
```
Theoretically, couldn't the WOLD_ID be retrieved from the concept list? https://github.com/concepticon/concepticon-data/blob/master/concepticondata/conceptlists/Haspelmath-2009-1460.tsv

tresoldi replied:
It could, but should we add them? They are not needed for lexibank (by analogy with the language ids), join operations can be carried with pyconcepticon. But I can add them, anybody else on board?

chrzyki replied:
No, I just thought that would make the lookup/mapping easier.

Commit:
Preparing for merging (pylexibank2 should be complete)

tresoldi commented:
Pushing updated version. As for feedback:

@SimonGreenhill I've fixed the Tzotzil glottocode
@chrzyki I didn't mention in the previous message that the concept list in Concepticon is missing a number of entries; I decided to read from there nonetheless, and only supplement the missing ones. The code should be more readable now.
@LinguList I've added an `etc/languages.csv` file and reworked the IDs of languages and concepts; please check if they are right.

SimonGreenhill reviewed (approved):
Looks good, thanks @tresoldi !

tresoldi commented:
Just to remember the two main issues here:

- the concept list ("Haspelmath-2009-1460") is not complete (it looks like only concepts mapped to Concepticon are there); missing concepts are currently loaded from the raw `parameters.csv`, but I'd prioritize mapping them (I'll see if I can do it today, at least in part, as this influences CLICS)

- the download function is not working (I don't know how to extract the data with the new library -- could do it by hand)

tresoldi commented:
Just counted, there about ~400 unmapped concepts.

xrotwang commented:
@tresoldi what exactly isn't working in the download method?

tresoldi commented:
@xrotwang it is my fault, but I didn't find the way to download and extract a ZIP file. Of course I could do it "by hand", but there was a method for that, right?

xrotwang commented:
@tresoldi but that's already done by `providers.clld`, see https://github.com/lexibank/pylexibank/blob/6bb8ae5a1324ecb9107267e122ed2cee44413d31/src/pylexibank/providers/clld.py#L16-L21

xrotwang commented:
Regarding the unmapped concepts: That's the 354 non-core meanings listed here https://wold.clld.org/meaning right? I'm inclined to leave them unmapped, mainly because of the very small coverage. (See the "Representation" column in the web app - the most covered non-core meanings are represented in just 4 languages.)

tresoldi commented:
@xrotwang So in this case I should just inherit from CLLD? What about .zip files which are not CLDF datasets?

As for the concepts, at least a few of them were added as concept sets to Concepticon since the first mapping, but I wonder if we shouldn't add them nonetheless to the conceptlist (without mapping, I mean).

xrotwang commented:
@tresoldi yes, inherit from `CLLD` - that was already the case before pylexibank 2.0.

xrotwang commented:
Rearding the non-core meanings: I think this is a trade-off, and a case of diminishing returns. Spending effort on including more meanings would only be justified if we strive for completeness - in which case including just a few more would not make sense.

xrotwang commented:
@tresoldi the 354 non-core meanings will contribute about 500 lexemes, less than 1% of the total - and many of these unmapped, or mapped to rather obscure concepts.

Commit:
Using CLLD provider

tresoldi commented:
@xrotwang converted to CLLD and tested the download function, it is working.

As for the non-core meanings, that's perfect, let's keep it that way.

LinguList commented:
Yes exactly, didn't we already discuss this a long time ago? Having just a few concepts with a skewed distribution in forms more won't really make a difference, and it also destroys the unity of how the 1430 concepts of WOLD are perceived. At least in their publications, they never considered more.
